sample-app-jsaddle/Material/ImageList.hs

```diff
@@ -0,0 +1,170 @@
+        , additionalAttributes :: [Miso.Attribute msg]
+        }
+
+
+{-| Default configuration of an image list
+-}
+config :: Config msg
+config =
+    Config
+        { masonry = False
+        , withTextProtection = False
+        , additionalAttributes = []
+        }
+
+
+{-| Specify whether an image list is a _masonry image list_
+The masonry image list variant presents images vertically arranged into several
+columns. In this layout, images may be any combination of aspect ratios.
+-}
+setMasonry :: Bool -> Config msg -> Config msg
+setMasonry masonry config_ =
+    config_ { masonry = masonry }
+
+
+{-| Specify whether an image list item's label should display in a scrim on top
+of the image
+By default, image list item's labels display below the image.
+-}
+setWithTextProtection :: Bool -> Config msg -> Config msg
+setWithTextProtection withTextProtection config_ =
+    config_ { withTextProtection = withTextProtection }
+
+
+{-| Specify additional attributes
+-}
+setAttributes :: [Miso.Attribute msg] -> Config msg -> Config msg
+setAttributes additionalAttributes config_ =
+    config_ { additionalAttributes = additionalAttributes }
+
+
+{-| Image list view function
+-}
+imageList :: Config msg -> [ImageListItem.ImageListItem msg] -> Miso.View msg
+imageList (config_@Config { additionalAttributes=additionalAttributes }) listItems =
+    Miso.nodeHtml "mdc-image-list"
+        (Maybe.mapMaybe id
+            [ rootCs
+            , masonryCs config_
+            , withTextProtectionCs config_
+            ]
+            ++ additionalAttributes
+        )
+        (Prelude.map (listItemElt config_) listItems)
+
+
+rootCs :: Maybe (Miso.Attribute msg)
+rootCs =
+    Just (Miso.class_ "mdc-image-list")
+
+
+masonryCs :: Config msg -> Maybe (Miso.Attribute msg)
+masonryCs (Config { masonry=masonry }) =
+    if masonry then
+        Just (Miso.class_ "mdc-image-list--masonry")
+
+    else
+        Nothing
+
+
+withTextProtectionCs :: Config msg -> Maybe (Miso.Attribute msg)
+withTextProtectionCs (Config { withTextProtection=withTextProtection }) =
+    if withTextProtection then
+        Just (Miso.class_ "mdc-image-list--with-text-protection")
+
+    else
+        Nothing
+
+
+listItemElt :: Config msg -> ImageListItem.ImageListItem msg -> Miso.View msg
+listItemElt (config_@Config { masonry=masonry }) (listItem@(ImageListItem.ImageListItem (ImageListItem.Config { ImageListItem.href=href, ImageListItem.additionalAttributes=additionalAttributes }))) =
+    let
+        inner =
+            [ if masonry then
+                imageElt masonry listItem
+
+              else
+                imageAspectContainerElt masonry listItem
+            , supportingElt listItem
+            ]
+    in
+    Miso.nodeHtml "mdc-image-list-item"
+        (Miso.class_ "mdc-image-list__item" : additionalAttributes)
+        (href
+            |> Maybe.maybe Nothing (\href_ -> Just [ Miso.a_ [ Miso.href_ (Miso.String.toMisoString href_) ] inner ])
```

Comment:
A suggestion that might be closer to the original:

```suggestion
            |> fmap (\href_ -> [ Miso.a_ [ Miso.href_ (Miso.String.toMisoString href_) ] inner ] )
```

The original code is:
```
|> Maybe.map (\href_ -> [ Html.a [ Html.Attributes.href href_ ] inner ])
```

Reply:
Moreover, It might be safe to replace `Maybe.map` with `fmap` using the script.

Reply:
It's now part of the script.
https://github.com/developandplay/elm-to-miso/commit/d258561b05f3bf1d8d3a64436edb2ad3e70ed66e